src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp

```diff
@@ -59,8 +58,12 @@ QUANTUM_SIMULATOR LoadQuantumSimulator()
     handle = ::dlopen(FULLSTATESIMULATORLIB, RTLD_LAZY);
     if (handle == nullptr)
     {
-        throw std::runtime_error(
-            std::string("Failed to load ") + FULLSTATESIMULATORLIB + " (" + ::dlerror() + ")");
+        handle = ::dlopen(XPLATFULLSTATESIMULATORLIB, RTLD_LAZY);
+        if (handle == nullptr)
+        {
+            throw std::runtime_error(
+                std::string("Failed to load ") + XPLATFULLSTATESIMULATORLIB + " (" + ::dlerror() + ")");
+        }
     }
 #endif

```

Comment:
(Just curiosity, feel free to ignore)
I have an impression that after this change the dynamic library `Microsoft.Quantum.Simulator.Runtime` gets the extension `.dll` _on all the platforms_.
But on non-Win platforms if the `.dll` library opening/loading failed, then 
Mac will try to load/open `.dylib`,
and the platforms other than Win and Mac will try to load/open `.so`.

If that's the case then why not leave it as it was
or why not make a full transition to `.dll` extension?

Reply:
This change was introduced by this [PR](https://github.com/microsoft/qsharp-runtime/pull/760).

The reason for this change was that in the package that contains the `Microsoft.Quantum.Simulator.Runtime` library, the library is always named as a DLL even if it is a Mac or Linux library, while local development and cross-platform pipelines need to handle the specific library name. For these two to co-exist happily, we first try load a .dll (which handles the case when a package reference to `Microsoft.Quantum.Simulators` is used), and if that fails on non-Windows platforms we try the native library name (which handles cases in local development and cross-platform pipelines).